### PR TITLE
Warns if variable name is composed only of underscores and digits.

### PR DESCRIPTION
**Pull request by AlexEne**
This implements: https://github.com/rust-lang-nursery/rust-clippy/issues/2182

Sample input:
```let __1___2 = 12 ```

Warning output:
```
error: binding whose name is just underscores and digits
   --> $DIR/non_expressive_names.rs:141:9
    |
141 |     let __1___2 = 12; //~ERROR Consider a more descriptive name
    |         ^^^^^^^
```

### CODE REVIEW

**Comment by AlexEne** [comment]
One thing that is unclear to me is how to update the documentations for errors. 
Most of them provide links to explanations of why this is bad. On the other hand this is kind of obviously wrong (regardless of language) so maybe it doesn't need such an addition.

File: clippy_lints/src/non_expressive_names.rs [hunk 1]
```diff
@@ -133,6 +153,15 @@ impl<'a, 'tcx, 'b> SimilarNamesNameVisitor<'a, 'tcx, 'b> {
         if interned_name.chars().any(char::is_uppercase) {
             return;
         }
+        if interned_name.chars().all(|c| c.is_digit(10) || c == '_') {
+            span_lint(
+                self.0.cx,
+                JUST_UNDERSCORES_AND_NUMBERS,
```
Comment (oli-obk) [inline clippy_lints/src/non_expressive_names.rs]: this should be `JUST_UNDERSCORES_AND_DIGITS`.

You can run clippy's tests with `cargo clippy`

File: tests/ui/non_expressive_names.rs [hunk 2]
```diff
@@ -134,3 +134,10 @@ fn bla() {
         }
     }
 }
+
+fn underscores_and_numbers() {
+    let _1 = 1; //~ERROR Consider a more descriptive name
+    let ____1 = 1; //~ERROR Consider a more descriptive name
+    let __1___2 = 12; //~ERROR Consider a more descriptive name
+    let _1_ok= 1;
+}
```
Comment (oli-obk) [inline tests/ui/non_expressive_names.rs]: missing newline at end of file

File: clippy_lints/src/non_expressive_names.rs [hunk 3]
```diff
@@ -133,6 +153,15 @@ impl<'a, 'tcx, 'b> SimilarNamesNameVisitor<'a, 'tcx, 'b> {
...
+                span,
+                "binding whose name is just underscores and digits",
```
Comment (oli-obk) [inline clippy_lints/src/non_expressive_names.rs]: add some suggestion on what to do, too. Maybe like the other message "consider choosing a more descriptive name"

**Comment by oli-obk** [comment]
> On the other hand this is kind of obviously wrong (regardless of language) so maybe it doesn't need such an addition.

Yea the docs are fine as you wrote them. I also think they need no further explanation

File: clippy_lints/src/non_expressive_names.rs (hunk 1 above)
Comment (AlexEne) [reply clippy_lints/src/non_expressive_names.rs]: Nice catch, I thought that they are includded in cargo test (or maybe I missed the output)

Comment (oli-obk) [reply clippy_lints/src/non_expressive_names.rs]: sorry, I meant `cargo test`..

Also, I didn't catch it, travis did ;)

**Comment by AlexEne** [comment]
I will do the changes tomorrow and update this

File: tests/ui/non_expressive_names.rs (hunk 2 above)
Comment (AlexEne) [reply tests/ui/non_expressive_names.rs]: I am super confused about this one, I did change it and it has an empty line on my machine, git diff shows nothing. I will figure it out tomorrow

**Comment by oli-obk** [comment]
Thanks!

**Pull request merged**